test/python/test_instructions.py

```diff
@@ -0,0 +1,353 @@
+            test_hwm),
+        navigate=lambda: navigator.navigate_and_compare(
+            TESTS_ROOT_DIR,
+            test_name,
+            instructions))
+
+    account.check_public_key(public_key)
+
+
+@pytest.mark.parametrize("account", [DEFAULT_ACCOUNT])
+def test_get_main_hwm(
+        account: Account,
+        client: TezosClient,
+        firmware: Firmware,
+        navigator: Navigator) -> None:
+    """Test the QUERY_MAIN_HWM instruction."""
+
+    main_chain_id: int = 0
+    main_hwm = Hwm(0)
+    test_hwm = Hwm(0)
+
+    instructions = get_setup_app_context_instructions(firmware)
+
+    send_and_navigate(
+        send=lambda: client.setup_app_context(
+            account,
+            main_chain_id,
+            main_hwm,
+            test_hwm),
+        navigate=lambda: navigator.navigate(instructions))
+
+    received_main_hwm = client.get_main_hwm()
+
+    assert received_main_hwm == main_hwm, \
+        f"Expected main hmw {main_hwm} but got {received_main_hwm}"
+
+
+@pytest.mark.parametrize("account", [DEFAULT_ACCOUNT])
+def test_get_all_hwm(
+        account: Account,
+        client: TezosClient,
+        firmware: Firmware,
+        navigator: Navigator) -> None:
+    """Test the QUERY_ALL_HWM instruction."""
+
+    main_chain_id: int = 0
+    main_hwm = Hwm(0)
+    test_hwm = Hwm(0)
+
+    instructions = get_setup_app_context_instructions(firmware)
+
+    send_and_navigate(
+        send=lambda: client.setup_app_context(
+            account,
+            main_chain_id,
+            main_hwm,
+            test_hwm),
+        navigate=lambda: navigator.navigate(instructions))
+
+    received_main_chain_id, received_main_hwm, received_test_hwm = client.get_all_hwm()
+
+    assert received_main_chain_id == main_chain_id, \
+        f"Expected main chain id {main_chain_id} but got {received_main_chain_id}"
+
+    assert received_main_hwm == main_hwm, \
+        f"Expected main hmw {main_hwm} but got {received_main_hwm}"
+
+    assert received_test_hwm == test_hwm, \
+        f"Expected test hmw {test_hwm} but got {received_test_hwm}"
+
+
+@pytest.mark.parametrize("account", [DEFAULT_ACCOUNT])
+@pytest.mark.parametrize("with_hash", [False, True])
+def test_sign_message(
+        account: Account,
+        with_hash: bool,
+        client: TezosClient,
+        firmware: Firmware,
+        navigator: Navigator) -> None:
+    """Test the SIGN(_WITH_HASH) instruction."""
+
+    main_chain_id: int = 0
+    main_hwm = Hwm(0)
+    test_hwm = Hwm(0)
+
+    instructions = get_setup_app_context_instructions(firmware)
+
+    send_and_navigate(
+        send=lambda: client.setup_app_context(
+            account,
+            main_chain_id,
+            main_hwm,
+            test_hwm),
+        navigate=lambda: navigator.navigate(instructions))
+
+    tx = RawMessage("0270000007000000000000000000000000000000000000000000000000000000000000000014000000000003000000010000000000000000000000000000000000000000000000000000000000000000")
```

Comment:
A commentt explaining what this raw msg is , which operation it is or how it is generated would be good.

Reply:
I had a way to generate them.